### ISSUE TITLE
[java] AbstractClassWithoutAnyMethod: False Positive on derived abstract class

### ISSUE THREAD

**Issue by mebigfatguy** [pmd/pmd#6965]
**Affects PMD Version:** 
7.23.0
**Rule:** [AbstractClassWithoutAnyMethod](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#abstractclasswithoutabstractmethod)
**Description:**

Class is indeed an empty abstract class that derives from another abstract class that does have abstract methods

**Code Sample demonstrating the issue:**

```java
public abstract class BaseFooType {
       public abstract Foo getFoo();
}

public abstract SpecificBoo extends BaseFooType {
}
```

**Expected outcome:**
no report
cant change to non abstract, nor can you add a private ctor
PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** Ant

**Comment by UncleOwen** [comment]
Sounds like a duplicate of https://github.com/pmd/pmd/issues/1237

**Comment by AzazelSensei** [comment]
Opened https://github.com/pmd/pmd/pull/6972 for this.
